core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -1201,4 +1223,74 @@ private boolean checkUniqueIdColumnInTable() throws UserStoreException {
         }
         return false;
     }
+
+    /**
+     * Delete tenant UM Data.
+     *
+     * @param tenantId Id of the tenant
+     * @param conn DB connection
+     * @throws Exception
+     */
+    private static void deleteTenantUMData(int tenantId, Connection conn) throws Exception {
+
+        try {
+            conn.setAutoCommit(false);
+
+            executeDeleteQuery(conn, DBConstants.DELETE_USER_PERMISSIONS_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, DBConstants.DELETE_ROLE_PERMISSIONS_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, DBConstants.DELETE_PERMISSION_BY_TENANT_ID_SQL, tenantId);
+
+            executeDeleteQuery(conn, ProfileDBConstant.DELETE_CLAIM_BEHAVIOR_BY_TENANT_ID, tenantId);
+            executeDeleteQuery(conn, ProfileDBConstant.DELETE_PROFILE_CONFIG_BY_TENANT_ID_SQL, tenantId);
+
+            executeDeleteQuery(conn, ClaimDBConstants.DELETE_CLAIM_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, ClaimDBConstants.DELETE_DIALECT_BY_TENANT_ID_SQL, tenantId);
+
+            executeDeleteQuery(conn, JDBCRealmConstants.DELETE_USER_PROPERTY_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, JDBCRealmConstants.DELETE_DOMAIN_FROM_USER_ROLE_BY_TENANT_ID, tenantId);
+            executeDeleteQuery(conn, JDBCRealmConstants.REMOVE_USER_ROLES_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, JDBCRealmConstants.DELETE_ROLES_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, JDBCRealmConstants.DELETE_USERS_BY_TENANT_ID_SQL, tenantId);
+
+            executeDeleteQuery(conn, HybridJDBCConstants.DELETE_ROLES_BY_TENANT_ID_SQL, tenantId);
+            executeDeleteQuery(conn, HybridJDBCConstants.DELETE_REMEMBERME_VALUES_BY_TENANT_ID_SQL, tenantId);
+
+            executeDeleteQuery(conn, TenantConstants.DELETE_TENANT_SQL, tenantId);
+
+            conn.commit();
+        } catch (Exception e) {
```

Comment:
Is it possible to catch a specific exception here rather than catching the Exception class?

Reply:
Here the SQLException is handled inside `executeDeleteQuery` method and it returns new Exception with the error message. So I thought to catch that `Exception` here. And both are private methods.
Any suggestions to handle them in a better way?